### DISCUSSION TITLE
Documentation: Add missing items to the Architecture page

### DISCUSSION THREAD

**Discussion by gziolo** [WordPress/gutenberg#49442] [Developer Experience]
In the block editor handbook, some initial changes were applied in #49184 to the current structure of [Architecture page](https://developer.wordpress.org/block-editor/explanations/architecture/) after the feedback received from @Mamaduka. 

What is missing on this page? 

<img width="646" alt="Screenshot 2023-03-29 at 15 08 51" src="https://user-images.githubusercontent.com/699132/228546265-c57cda29-0130-49c9-9507-8b8fce149093.png">

---

It's also an excellent opportunity to discover missing key concepts related to the Gutenberg project. We have the following covered:

- Blocks
- Composability
- Data & Attributes
- Block Transforms
- Block Variations
- Reusable Blocks
- Patterns
- Templates
- Styles

**Comment by Mamaduka** [comment]
I think these two items would be a great addition to the page.

* RichText - I can't find any docs about architecture/design.
* History - how the editor records "edit" steps and handles undo/redo actions.

P.S. I'm planning to tackle the issues for [the "History" feature](https://github.com/WordPress/gutenberg/labels/%5BFeature%5D%20History). So I'll be able to write that page myself eventually, but having more context about previous decisions is always appreciated.

**Comment by juanmaguitar** [comment]
@ndiego

**Comment by talldan** [comment]
My thoughts, it could be an idea to structure this in terms of:
- Lead with a section for the pure block editor architecture, excluding WordPress-specific concepts.
- Top-level items about the WordPress implementation of the block editor (Data/Entities, our History implementation, Openverse?, general notes on extensibility)
- Project structure
- Some specific notes on each editor
    - Site editor (Templates, Template Parts, Styles, Layout, Query loop, Post Templates, Extensibility)
    - Post Editor (Post types, Extensibility)
    - Widget Editors (Sidebars, Block Widgets, Customizer integration)
